### PR TITLE
Add adoption metrics CLI

### PR DESCRIPTION
**Pull request by graphanov**
## Summary
- Adds `osc metrics` with human table output, `--json`, `--since`, `--lookback`, `--table`, and `--verbose`.
- Computes local scaffold health signals: plan distribution, cycle time, stale active plans, close velocity, evidence completeness, and approval distribution.
- Documents the metrics command in the workflow guide, closes plan 059, and prepares `open-scaffold@0.4.11` as an owner-gated package candidate.

## Traceability
- Opened/advanced by Open Scaffold runner automation.
- Selected source: `backlog_plan` — `.osc/plans/done/059-adoption-metrics.md`.
- Evidence: `.osc/releases/2026-05-21-059-adoption-metrics.md`.
- GitHub issue: N/A; no open issue candidate existed for this selected slice.

## Verification
- `git diff --check` — pass.
- `npm test -- --run tests/metrics.test.ts` — pass; 1 file / 8 tests.
- `npm test -- --run` — pass; 32 files / 280 tests.
- `npm run build` — pass.
- `node dist/cli.js metrics` — pass; default human-readable metrics table renders without flags.
- `node dist/cli.js metrics --json | jq .summary.total_plans` — pass; returned `89`.
- `node dist/cli.js metrics --json | jq .cycle_time.mean_days` — pass; returned `0.99`.
- `node dist/cli.js metrics --lookback 4 --verbose` — pass.
- `npm pack --dry-run --json` — pass; candidate `open-scaffold@0.4.11` includes `dist/metrics.js` and `dist/metrics.d.ts`.
- `./verify.sh --strict` — pass; 10 pass / 0 fail / 0 warn.

## Review notes
- Codex round 1 actionable findings addressed: `osc metrics` now runs with no flags, and mtime fallback cycle-time samples are included for done plans without evidence-note dates.
- Codex round 2 actionable findings addressed: `last month` now clamps month-end dates to the previous month, and evidence notes now match exact plan-slug boundaries instead of substring matches.
- Codex round 3 actionable findings addressed: invalid ISO datetime filters are rejected, and direct evidence file matches now take priority over newer text-only mentions.
- Latest-head Codex re-review requested after the final fix commit.

## Owner gates
- Merge remains owner-gated.
- npm publication remains owner-gated.
- GitHub Release creation/latest movement remains owner-gated.

## Out of scope
- No charts, dashboard, metrics snapshots, database, analytics service, network collector, runtime spawning, npm publish, GitHub Release mutation, or merge automation.

### CODE REVIEW

**Comment by graphanov** [comment]
@codex review

Please review PR #86 latest head 7e7a14ab0cee9b0052d3a95a9c5d6af07c11fb8c for actual diff correctness. Report any actionable issues or confirm none.

File: src/cli.ts [hunk 1]
```diff
@@ -1392,6 +1394,87 @@ function evolutionCommand(args: string[]): void {
...
+    console.error(`Missing value for ${flag}`);
+    process.exit(2);
+  }
+  return value;
+}
+
+function metricsCommand(args: string[]): void {
+  if (args[0] === undefined || isHelpArg(args[0])) {
+    printMetricsUsage('stdout');
+    return;
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/cli.ts]: Fixed in 2d72b1d. `osc metrics` now treats only explicit help flags as usage requests, so the no-argument command renders the default metrics table.

Verification:
- `npm test -- --run tests/metrics.test.ts` — pass
- `npm test -- --run` — pass; 32 files / 278 tests
- `npm run build` — pass
- `node dist/cli.js metrics` — pass
- `./verify.sh --strict` — pass; 10 pass / 0 fail / 0 warn

File: src/metrics.ts [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+    const close = planClosedDate(root, plan, evidence, hasGit);
+    const ageDays = round(Math.max(0, daysBetween(plan.statMtime, now)), 1);
+    const stale = plan.stage === 'active' && ageDays > STALE_DAYS;
+    const hasEvidence = Boolean(evidence && evidence.text.trim().length > 0);
+    const approval = approvalFromEvidence(evidence);
+
+    if (plan.stage === 'done') {
+      if (close.date && close.date >= windowStart && close.date <= now) closedInWindow += 1;
+      if (close.date && close.source !== 'mtime' && close.date >= createdAt) {
+        cycleDurations.push(daysBetween(createdAt, close.date));
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/metrics.ts]: Fixed in 2d72b1d. Done plans using the mtime close fallback now contribute cycle-time samples instead of being marked insufficient, and the regression test covers the no-evidence mtime fallback case.

Verification:
- `npm test -- --run tests/metrics.test.ts` — pass
- `npm test -- --run` — pass; 32 files / 278 tests
- `npm run build` — pass
- `node dist/cli.js metrics --json | jq .cycle_time.mean_days` — pass; returned `0.99`
- `./verify.sh --strict` — pass; 10 pass / 0 fail / 0 warn

**Comment by graphanov** [comment]
@codex review

Please re-review PR #86 latest head 2d72b1d60aafc9851e3d5f756f62a956beaacd06 for actual diff correctness after the pushed fixes. Prior actionable findings have been addressed; report any remaining actionable issues or confirm none.

**Comment by graphanov** [comment]
@codex review

Please re-review PR #86 latest head 1b39e1d4a5c55af7ef2e7f22a6255a29da90f72a for actual diff correctness after the final pushed fixes. Prior actionable findings have been addressed with regression coverage; report any remaining actionable issues or confirm none.

File: src/metrics.ts [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+  }
+
+  const weeksAgo = raw.match(/^(\d+)\s+weeks?\s+ago$/i);
+  if (weeksAgo) {
+    return new Date(startOfUtcDay(now).getTime() - Number(weeksAgo[1]) * 7 * DAY_MS);
+  }
+
+  if (/^last\s+month$/i.test(raw)) {
+    const day = startOfUtcDay(now);
+    return new Date(Date.UTC(day.getUTCFullYear(), day.getUTCMonth() - 1, day.getUTCDate()));
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/metrics.ts]: Fixed in 1b39e1d4a5c55af7ef2e7f22a6255a29da90f72a. `last month` now computes the previous UTC month and clamps the day to that month’s last valid day, including leap-year and January/year-boundary cases.

Verification:
- `npm test -- --run tests/metrics.test.ts` — pass; 1 file / 7 tests
- `npm test -- --run` — pass; 32 files / 279 tests
- `npm run build` — pass
- `./verify.sh --strict` — pass; 10 pass / 0 fail / 0 warn

File: src/metrics.ts [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+    });
+}
+
+function escapeRegex(value: string): string {
+  return value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
+function evidenceForPlan(notes: EvidenceNote[], slug: string): EvidenceNote | null {
+  const slugPattern = new RegExp(`(^|[^A-Za-z0-9])${escapeRegex(slug)}([^A-Za-z0-9]|$)`);
+  const matches = notes.filter((note) => note.relativePath.includes(slug) || slugPattern.test(note.text));
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/metrics.ts]: Fixed in 1b39e1d4a5c55af7ef2e7f22a6255a29da90f72a. Evidence note path matching now compares either the exact note basename or a strict `YYYY-MM-DD-<slug>` basename after stripping the date prefix, and text matching treats hyphen-adjacent suffixes as non-matches. Added regression coverage so `foo`, `001-foo`, and `001-foo-bar` do not cross-match.

Verification:
- `npm test -- --run tests/metrics.test.ts` — pass; 1 file / 7 tests
- `npm test -- --run` — pass; 32 files / 279 tests
- `npm run build` — pass
- `./verify.sh --strict` — pass; 10 pass / 0 fail / 0 warn

File: src/metrics.ts [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+
+function evidenceTextMatchesSlug(text: string, slug: string): boolean {
+  const slugTokenPattern = new RegExp(`(^|[^A-Za-z0-9-])${escapeRegex(slug)}([^A-Za-z0-9-]|$)`);
+  return slugTokenPattern.test(text);
+}
+
+function evidenceForPlan(notes: EvidenceNote[], slug: string): EvidenceNote | null {
+  const matches = notes.filter((note) => evidencePathMatchesSlug(note.relativePath, slug) || evidenceTextMatchesSlug(note.text, slug));
+  if (matches.length === 0) return null;
+  return matches.sort((a, b) => (a.date?.getTime() ?? 0) - (b.date?.getTime() ?? 0)).at(-1) ?? null;
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/metrics.ts]: Fixed in eaa08ba104888f645972e93df683ee72797b6472. `evidenceForPlan` now evaluates direct filename matches first and only falls back to text mentions when there is no direct evidence file match, so newer cross-reference notes cannot override a plan’s own evidence note.

Verification:
- `npm test -- --run tests/metrics.test.ts` — pass; 1 file / 8 tests
- `npm test -- --run` — pass; 32 files / 280 tests
- `npm run build` — pass
- `./verify.sh --strict` — pass; 10 pass / 0 fail / 0 warn

File: src/metrics.ts [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+  const raw = value.trim();
+  if (!raw) throw new Error('Missing date value for --since');
+
+  const isoDateOnly = parseIsoDateOnly(raw);
+  if (isoDateOnly) return isoDateOnly;
+
+  if (/^\d{4}-\d{2}-\d{2}T/.test(raw)) {
+    const parsed = new Date(raw);
+    if (Number.isNaN(parsed.getTime())) throw new Error(`Invalid ISO 8601 date for --since: ${raw}`);
+    return parsed;
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/metrics.ts:152]: Fixed in eaa08ba104888f645972e93df683ee72797b6472. The ISO datetime branch now validates the calendar date prefix with the same strict date validation used by date-only input before accepting the parsed datetime, so normalized invalid dates such as `2026-02-30T00:00:00Z` are rejected.

Verification:
- `npm test -- --run tests/metrics.test.ts` — pass; 1 file / 8 tests
- `npm test -- --run` — pass; 32 files / 280 tests
- `npm run build` — pass
- `./verify.sh --strict` — pass; 10 pass / 0 fail / 0 warn

**Comment by graphanov** [comment]
@codex review

Please re-review PR #86 latest head eaa08ba104888f645972e93df683ee72797b6472 for actual diff correctness after the latest pushed fixes. All prior actionable findings have been addressed with regression coverage; report any remaining actionable issues or confirm none.
